compiler/src/formatting/fmt.re

```diff
@@ -2696,7 +2696,19 @@ let print_type = (fmt, {ptyp_desc, ptyp_loc}) => {
       )
       ++ break,
     )
-  | PTyArrow([{ptyp_arg_label: Unlabeled} as param], return) =>
+  | PTyArrow(
+      [
+        {
+          ptyp_arg_label: Unlabeled,
+          ptyp_arg_type: {
+            ptyp_desc:
+              PTyAny | PTyVar(_) | PTyArrow(_, _) | PTyConstr(_, _) |
+              PTyPoly(_, _),
```

Comment:
I would probably just keep this to `PTyAny`, `PTyVar`, and `PTyArrow`.

Reply:
I removed the `PtyPoly` and limited `PTyConstr` to not have any data attached, we need to allow `PtyConstr` still though because something like `Number` is a constructor.